source/opt/def_use_manager.cpp

```diff
@@ -36,7 +36,8 @@ void DefUseManager::AnalyzeInstDef(ir::Instruction* inst) {
   }
 }
 
-void DefUseManager::AnalyzeInstUse(ir::Instruction* inst) {
+void DefUseManager::AnalyzeInstUse(ir::Instruction* inst, bool is_new_inst) {
+  if (!is_new_inst) EraseUseRecordsOfOperandIds(inst);
```

Comment:
Instead of adding a new parameter, you can check if `used_ids` (defined below) is empty or not.  Make this more transparent to the user.

Reply:
Fixed